Sources/LiveKit/Core/Room.swift

```diff
@@ -19,26 +19,53 @@ import Network
 import Promises
 import WebRTC
 
-public class Room: MulticastDelegate<RoomDelegate> {
+@objc
+public class Room: NSObject, Loggable {
+
+    // MARK: - MulticastDelegate
+
+    private var delegates = MulticastDelegate<RoomDelegateObjC>()
 
     internal let queue = DispatchQueue(label: "LiveKitSDK.room", qos: .default)
 
     // MARK: - Public
 
+    @objc
     public var sid: Sid? { _state.sid }
+
+    @objc
     public var name: String? { _state.name }
+
+    @objc
     public var metadata: String? { _state.metadata }
+
+    @objc
     public var serverVersion: String? { _state.serverVersion }
+
+    @objc
     public var serverRegion: String? { _state.serverRegion }
 
+    @objc
     public var localParticipant: LocalParticipant? { _state.localParticipant }
+
+    @objc
     public var remoteParticipants: [Sid: RemoteParticipant] { _state.remoteParticipants }
+
+    @objc
     public var activeSpeakers: [Participant] { _state.activeSpeakers }
 
     // expose engine's vars
+    @objc
     public var url: String? { engine._state.url }
+
+    @objc
     public var token: String? { engine._state.token }
+
     public var connectionState: ConnectionState { engine._state.connectionState }
+
+    @objc(connectionState)
+    public var connectionStateObjC: ConnectionStateObjC { engine._state.connectionState.toObjCType() }
```

Comment:
Hi @piercifani , I'm being extra careful not to break current API so we can keep all Swift features.

For example for ObjC, there is a new `connectionStateObjC` property but this will be renamed to `connectionState` when compiled for ObjC. This will use the new ObjC compatible `ConnectionStateObjC` enum.
https://github.com/livekit/client-sdk-swift/pull/128/files#diff-78980460aeafb35975431b7cd2aa713d7c9e40b3bb56e58b31b40b48fd1ddef6R66-R67

From Swift, there will be 2 properties visible `connectionState` and `connectionStateObjC` which isn't great but at least this approach doesn't break the current `connectionState` property. (Still looking if I can hide `connectionStateObjC` when compiled for Swift)

So the only downside for this PR would be new `-ObjC` post fixed properties / methods for ObjC compatibility.

Did you find something in particular that breaks ?